### DISCUSSION TITLE
Focus order in Popover

### DISCUSSION THREAD

**Discussion by james-ob-nf** [radix-ui/primitives#1922] [Help]
I'm using the popover component to implement a popover menu, which consists of an ordered list of buttons and links. This menu is embedded within a table row, so there are multiple instances of the menu within the page. This works great, however when tabbing through an open menu the first item to come into focus is always the first button, whether it's the first item or not. I'd like the first item in the DOM to be selected.

Using `event.preventDefault()` gives me the outcome I'm looking for, but when using this we've noticed an issue on iPhone where an open menu won't close when another trigger in the table is clicked, and both menus remain open until focus is moved outside either of the menus (and not on a trigger).

Is there another way I can set focus on the first child item when a popover is opened?

**Comment by benoitgrelard** [comment]
> I'm using the popover component to implement a popover menu, which consists of an ordered list of buttons and links.

It sounds like you're after `DropdownMenu` perhaps?
